AMBARI-23084 Background Ops Modal Updates - Get user name

majorendre opened the pull request:
## What changes were proposed in this pull request?

Storing and retrieving user_name for RequestEntity.

## How was this patch tested?

[WIP]
JUnit tests passed.
Some manual test done on my local cluster. No upgrade test done, yet.

(Please explain how this patch was tested. Ex: unit tests, manual tests)
(If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

Please review [Ambari Contributing Guide](https://cwiki.apache.org/confluence/display/AMBARI/How+to+Contribute) before opening a pull request.

@swagle Please have a look.

Commit:
AMBARI-23084 Background Ops Modal Updates - Get user name

swagle commented on ambari-server/src/main/java/org/apache/ambari/server/actionmanager/Request.java:
```diff
@@ -96,6 +98,7 @@ public Request(@Assisted long requestId, @Assisted("clusterId") Long clusterId,
     this.endTime = -1;
     this.exclusive = false;
     this.clusterHostInfo = "{}";
+    this.userName = AuthorizationHelper.getAuthenticatedName();
```
Since this is a part of the constructor, care should be taken to allow for nulls for a background operation lets say that is not initiated by a user.

majorendre replied:
AuthorizationHelper.getAuthenticatedName() returned "internal" in such cases, as far as I could see. 
Should I just allow nulls?
Should I move the AuthorizationHelper.getAuthenticatedName() call out of the constructor?

swagle replied:
Oh, you were quick to get to this while I was still reviewing :-)
I think nulls are ok, maybe other reviewers can chime in.

swagle commented on ambari-server/src/main/java/org/apache/ambari/server/orm/entities/RequestEntity.java:
```diff
@@ -125,6 +125,9 @@
   @Column(name = "exclusive_execution", insertable = true, updatable = true, nullable = false)
   private Integer exclusive = 0;
 
+  @Column(name = "user_name", nullable = false)
```
We do set user of a BatchRequest as "internal" so this non-nullable might be ok but make sure to check other scenarios for this, IMO enforcing this as being non-nullable should not be needed.

majorendre replied:
Thanks. I'll modify it tomorrow.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/831/
Test FAILed.
Test FAILured.

Commit:
Revert "AMBARI-23084 Background Ops Modal Updates - Get user name"

This reverts commit 0651dd808f3c71dcc152b3e85f7c4048ac142724.

Commit:
AMBARI-23084 Background Ops Modal Updates - Get user name

2nd edition

majorendre commented:
Reverted my initial commit, and my last commit contains all the changes.
Updated according the the comments. 
I guess, the failed tests above have nothing to do with me.

adoroszlai commented on ambari-server/src/main/java/org/apache/ambari/server/controller/internal/RequestStageContainer.java:
```diff
@@ -216,6 +217,10 @@ public void persist() throws AmbariException {
       }
 
+      if (request != null) { //request can be null at least in JUnit
+        request.setUserName(AuthorizationHelper.getAuthenticatedName());
```
```
[ERROR]   RequestStageContainerTest.testPersist:152
  Unexpected method call Request.setUserName(null)
```

adoroszlai reviewed (changes requested):
LGTM, except one failing unit test.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/840/
Test FAILed.
Test FAILured.

Commit:
AMBARI-23084 mended failing junit

majorendre commented:
Mended the failing unit test.

jonathan-hurley commented on ambari-server/src/main/java/org/apache/ambari/server/actionmanager/Request.java:
```diff
@@ -422,6 +424,14 @@ public void setExclusive(boolean isExclusive) {
+  public void setUserName(String userName) {
+    this.userName = userName;
+  }
+
```
Javadoc.

jonathan-hurley commented on ambari-server/src/main/java/org/apache/ambari/server/orm/entities/RequestEntity.java:
```diff
@@ -299,6 +302,14 @@ public void setRequestScheduleId(Long scheduleId) {
+  public void setUserName(String userName) {
+    this.userName = userName;
+  }
+
```
Javadoc.

jonathan-hurley reviewed (approved):
Just some missing JavaDoc.

Commit:
AMBARI-23084 javadoc added

majorendre commented:
Added som javadoc.

swagle commented on ambari-server/src/main/java/org/apache/ambari/server/controller/internal/RequestResourceProvider.java:
```diff
@@ -851,6 +853,7 @@ private Resource getRequestResource(RequestEntity entity, String clusterName,
             hostRoleStatusCounters.get(HostRoleStatus.QUEUED), requestedPropertyIds);
     setResourceProperty(resource, REQUEST_COMPLETED_TASK_CNT_ID,
             hostRoleStatusCounters.get(HostRoleStatus.COMPLETED), requestedPropertyIds);
+    setResourceProperty(resource, REQUEST_USER_NAME, entity.getUserName(), requestedPropertyIds);
```
UI should make sure to handle null as something like Unknown user for readability.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/845/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/846/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/847/
Test FAILed.
Test FAILured.
